mindsdb_sql/parser/dialects/mindsdb/parser.py

```diff
@@ -99,18 +101,21 @@ def create_kb(self, p):
         from_query = getattr(p, 'select', None)
         name = p.identifier
         # check model and storage are in params
-        model = params.pop('model', None) or params.pop('MODEL', None)  # case insensitive
-        storage = params.pop('storage', None) or params.pop('STORAGE', None)  # case insensitive
+        params = {k.lower(): v for k, v in params.items()}  # case insensitive
+        model = params.pop('model', None)
+        storage = params.pop('storage', None)
+
+        if not storage:
+            # convert to identifier
+            if isinstance(model, str):
+                storage = Identifier(storage)
```

Comment:
I think something wrong here: if not storage: use it

Reply:
good, point. fixed